### ISSUE TITLE
eslint/tslint behavior difference with comma-dangle

### ISSUE THREAD

**Issue by danvk** [palantir/tslint#1122]
### Bug Report (delete this for feature requests)

- __TSLint version__: `3.7.3`
- __TypeScript version__: `1.8.9`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function f() {
  return [1, {
    a: 2,
    b: 2,
  }];
}
```

with `tslint.json`:

```json
{
  "rules": {
    "trailing-comma": [
      true,
      {
        "multiline": "always",
        "singleline": "never"
      }
    ]
  }
}
```

#### Actual behavior

```
repro.ts[5, 3]: missing trailing comma
```

#### Expected behavior

(no error)

When I run this code through eslint with the following config:

```javascript
{
  "rules": {
    'comma-dangle': [2, 'always-multiline'],
  }
}
```

it passes. But tslint wants me to write this instead:

```ts
function f() {
  return [1, {
    a: 2,
    b: 2,
  },];  // <---
}
```

which I don't think makes much sense. In fact, eslint complains about it:

```
repro.ts
  5:4  error  Unexpected trailing comma  comma-dangle
```

**Comment by unional** [comment]
Exactly the issue I have with this: https://github.com/palantir/tslint/blob/3.7.4/src/configuration.ts#L52

**Comment by adidahiya** [comment]
fair enough. we'd be open to a PR that adjusts the behavior here to align with ESLint.

**Comment by JKillian** [comment]
Hi all, please see #1145 and check if it does what you need.

**Comment by unional** [comment]
Interesting. and 👍
